### PR TITLE
Seedv2: More simple version to seedv1 fixes #585 and #618

### LINKED ISSUE #585: Reproductibility for sampler => Define a global DDESEED
Hi, 
I figured out that 

https://github.com/lululxvi/deepxde/blob/707b0984312d0ad5f28522b2aa863fed923b424a/deepxde/geometry/sampler.py#L28

was giving different data points despite having previously defined ``set_random_seed``. I didn't notice this before as I was using `Rectangle` with `uniform` generator.

I obtain the same results if I set a proper seed here, e.g.,

```
seed = 11
set_random_seed(seed)
```

and 

``
np.random.default_rng(seed).
``

So far, I didn't test with other random generators. I also noticed that seeds appeared for the initializers: 

https://github.com/lululxvi/deepxde/blob/707b0984312d0ad5f28522b2aa863fed923b424a/deepxde/nn/initializers.py#L29

Yet, I didn't have issues with that. Initializers seem to be within the ``set_random_seed(seed)`` domain.

To properly set up everything, I propose to set a new global variable ``DDESEED`` with default value `None`. Then, if `DDESEED` is an integer, we would ``set_random_seed(seed)`` and call the seed value in sampler.py.

If you are ok with the idea, I will work on a PR, best!

### PR DESCRIPTION
**Pull request by pescap**
Hi,
I started from the beginning #599 to solve #585 and #618

### CODE REVIEW

File: deepxde/config.py [hunk 1]
```diff
@@ -6,6 +7,8 @@
 from .backend import backend_name, tf, torch
 from .real import Real
 
+thismod = sys.modules[__name__]
+setattr(thismod, "random_seed", None)
```
Comment (lululxvi) [inline deepxde/config.py:11]: Directly use
```python
random_seed = None
```

File: deepxde/config.py [hunk 2]
```diff
@@ -58,3 +61,5 @@ def set_random_seed(seed):
     elif backend_name == "jax":
         global jax_random_seed
         jax_random_seed = seed
+    thismod = sys.modules[__name__]
+    setattr(thismod, "random_seed", seed)
```
Comment (lululxvi) [inline deepxde/config.py:65]:
```python
global random_seed
random_seed = seed
```

File: deepxde/config.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import sys
```
Comment (lululxvi) [inline deepxde/config.py:1]: No need to import sys

**Comment by pescap** [comment]
Tested successfully with: 
```
"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch"""
import deepxde as dde

dde.config.set_random_seed(10)

def pde(x, y):
    dy_xx = dde.grad.hessian(y, x, i=0, j=0)
    dy_yy = dde.grad.hessian(y, x, i=1, j=1)
    return -dy_xx - dy_yy - 1


def boundary(_, on_boundary):
    return on_boundary


geom = dde.geometry.Polygon([[0, 0], [1, 0], [1, -1], [-1, -1], [-1, 1], [0, 1]])
bc = dde.icbc.DirichletBC(geom, lambda x: 0, boundary)

data = dde.data.PDE(geom, pde, bc, num_domain=1200, num_boundary=120, num_test=1500)
net = dde.nn.FNN([2] + [50] * 4 + [1], "tanh", "Glorot uniform")
model = dde.Model(data, net)

model.compile("adam", lr=0.001)
model.train(epochs=2000)
```
